Source/Core/Core/HW/DVDInterface.cpp

```diff
@@ -1244,30 +1253,28 @@ void ScheduleReads(u64 dvd_offset, u32 dvd_length, bool decrypt, u32 output_addr
             buffer_start, buffer_end, buffer_end - buffer_start);
 
   DEBUG_LOG(DVDINTERFACE,
-            "Schedule reads: offset=0x%" PRIx64 " length=0x%" PRIx32 " address=0x%" PRIx32,
-            dvd_offset, dvd_length, output_address);
+            "Schedule reads: offset=0x%" PRIx64 " length=0x%" PRIx32 " address=0x%" PRIx32, offset,
+            length, output_address);
 
   // The DVD drive's minimum turnaround time on a command, based on a hardware test.
   s64 ticks_until_completion = COMMAND_LATENCY_US * (SystemTimers::GetTicksPerSecond() / 1000000);
 
   u32 buffered_blocks = 0;
   u32 unbuffered_blocks = 0;
 
-  while (dvd_length > 0)
+  while (length > 0)
   {
-    // Where the read actually takes place on disc
-    u64 rounded_offset = Common::AlignDown(dvd_offset, DVD_ECC_BLOCK_SIZE);
+    const u32 bytes_per_chunk =
+        decrypt ? DiscIO::CVolumeWiiCrypted::BLOCK_DATA_SIZE : DVD_ECC_BLOCK_SIZE;
```

Comment:
I realize now that the `ticks_until_completion` calculation shouldn't be using `chunk_length`, but rather the actual bytes read from the DVD. My guess for how it works on hardware is that IOS will read the whole block from DI the first time the PPC wants to read it, and then keep it in an IOS-internal cache as long as PPC doesn't request another block. It's only a guess, though. Is that worth implementing? The difference won't be especially big, and likely not bigger than the time it takes to decrypt. (I don't know how long it takes to decrypt.)